kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyReconciler.java

```diff
@@ -399,6 +400,11 @@ private static ConfigurationFragment<Optional<TrustProvider>> buildTrustProvider
     @Override
     public UpdateControl<KafkaProxy> reconcile(KafkaProxy primary,
                                                Context<KafkaProxy> context) {
+        context.getSecondaryResource(Deployment.class, DEPLOYMENT_DEP).ifPresent(deployment -> {
+            Integer readyReplicas = deployment.getStatus().getReadyReplicas();
```

Comment:
The description in the CRD says `status.replicas` is the number of scheduled pods, but this seems to be the number of ready pods.

Also, maybe a null check on the status, just in case this manages to get called with a non-null deployment but null status.

Reply:
I've got some updated null handling locally as I fix the coverage issues.